src/main/java/hudson/plugins/ec2/ssh/EC2UnixLauncher.java

```diff
@@ -322,6 +333,36 @@ private File createIdentityKeyFile(EC2Computer computer) throws IOException {
         }
     }
 
+    private File createHostKeyFile(EC2Computer computer, String ec2HostAddress, TaskListener listener) throws IOException {
+        HostKey ec2HostKey = HostKeyHelper.getInstance().getHostKey(computer);
+        if (ec2HostKey == null){
+            return null;
+        }
+        File tempFile = File.createTempFile("ec2_", "_known_hosts");
+        String knownHost = "";
+        knownHost = String.format("%s %s %s", ec2HostAddress, ec2HostKey.getAlgorithm(), Base64.getEncoder().encodeToString(ec2HostKey.getKey()));
+
+        try {
+            FileOutputStream fileOutputStream = new FileOutputStream(tempFile);
```

Comment:
This bit will probably just read better in a try with resources way
try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(tempFile), StandardCharsets.UTF_8))

which will take care of closing it on an exception

Reply:
I've given this a shot. My java isn't great, so would appreciate a double check of my changes.